sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcTest.scala

```diff
@@ -78,12 +78,16 @@ abstract class OrcTest extends QueryTest with FileBasedDataSourceTest with Befor
       (f: String => Unit): Unit = withDataSourceFile(data)(f)
 
   /**
-   * Writes `data` to a Orc file and reads it back as a `DataFrame`,
+   * Writes `df` dataframe to a Orc file and reads it back as a `DataFrame`,
    * which is then passed to `f`. The Orc file will be deleted after `f` returns.
    */
-  protected def withOrcDataFrame[T <: Product: ClassTag: TypeTag]
-      (data: Seq[T], testVectorized: Boolean = true)
-      (f: DataFrame => Unit): Unit = withDataSourceDataFrame(data, testVectorized)(f)
+  protected def withOrcDataFrame(df: DataFrame, testVectorized: Boolean = true)
```

Comment:
@viirya why do we need to change this? Looks we can just add the overridden version to test nested DataFrame without touching other tests.

Reply:
I remember adding overridden version causes compilation error. Scala compiler complains ambiguity of overridden methods.

Reply:
Hm .. then I guess maybe it's good to just have a separate method for it. Most of changes here look caused by this one in the tests.

Reply:
I think it is for parity with the change at Parquet side (i.e. `ParquetTest`) in #27728. It should look better if Parquet and Orc have the same helper methods for test.

Reply:
Ok